File: LC/python/Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param a ListNode
    # @return a ListNode
    def swapPairs(self, head):
        if head == None or head.next == None:
            return head
        
        dummy = ListNode(0)
        dummy.next = head
        tmp = dummy
        
        while tmp.next and tmp.next.next:
            p = tmp.next.next
            tmp.next.next = p.next
            p.next = tmp.next
            tmp.next = p
            tmp = tmp.next.next
        
        return dummy.next
        

File: LC/java/Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null || head.next==null){
            return head;
        }
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode tmp = dummy;
        ListNode p;
        
        while(tmp.next!=null && tmp.next.next!=null){
            p = tmp.next.next;
            tmp.next.next = p.next;
            p.next = tmp.next;
            tmp.next = p;
            tmp=tmp.next.next;
        }
        return dummy.next;
    }
}